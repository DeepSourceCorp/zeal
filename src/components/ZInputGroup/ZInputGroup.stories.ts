import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'

import ZInputGroup from '@/components/ZInputGroup/ZInputGroup.vue'

export default {
  title: 'InputGroup',
  component: ZInputGroup,
  excludeStories: /.*Data$/
}

export const BasicInputGroupWithAddon = () => ({
  components: { ZInputGroup },
  data() {
    return {
      name: 'Hello World',
      sizes: ['x-small', 'small', 'medium', 'large', 'xlarge']
    }
  },
  template: `
    <div class='padded-container'>
      <div class="input-container">
        <z-input-group v-model="name" prefix="www." addon-bg-color="bg-juniper" />
      </div>
    </div>
      `
})

export const Sizes = () => ({
  components: { ZInputGroup },
  filters: {
    capitalize: function(value: string) {
      if (!value) {
        return ''
      }
      return (
        value
          .toString()
          .charAt(0)
          .toUpperCase() + value.slice(1)
      )
    }
  },
  data() {
    return {
      name: 'Hello World',
      sizes: ['x-small', 'small', 'medium', 'large', 'xlarge']
    }
  },
  template: `<div class='padded-container'>
        <div class="input-container space-y-4">
        <div v-for="size in sizes" :key="size" class="space-y-2">
          <h3 class="text-vanilla-100 text-xl font-semibold">{{ size | capitalize }}</h3>
          <z-input-group v-model="name" :size="size" prefix="www." addon-bg-color="bg-juniper" />
        </div>
          </div>
    </div>`
})
