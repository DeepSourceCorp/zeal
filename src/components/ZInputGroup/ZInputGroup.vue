<template>
  <div class="flex">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'ZInputGroup'
})
</script>
