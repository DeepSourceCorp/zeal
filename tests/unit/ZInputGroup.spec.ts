import { createLocalVue, mount, Wrapper } from '@vue/test-utils'
import Vue from 'vue'

import ZInputGroup from '../../src/components/ZInputGroup'

describe('InputGroup Component', () => {
  const localVue = createLocalVue()

  const mountFn = (options = {}) => {
    return mount(ZInputGroup, {
      localVue,
      stubs: {
        ZInput: true
      },
      ...options
    })
  }

  it('renders an input element with addon', () => {
    const wrapper = mountFn({
      propsData: {
        prefix: 'text-prefix'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders an input element with addon having a different background color', () => {
    const wrapper = mountFn({
      propsData: {
        prefix: 'text-prefix',
        addonBgColor: 'bg-juniper'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('accepts fallthrough attributes', () => {
    const wrapper = mountFn({
      propsData: {
        prefix: 'text-prefix',
        size: 'small'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
