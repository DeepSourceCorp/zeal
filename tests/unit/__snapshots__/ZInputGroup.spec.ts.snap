// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup Component accepts fallthrough attributes 1`] = `
<div class="flex" size="small">
  <div class="flex items-center p-2 rounded-tl-sm rounded-bl-sm outline-none bg-ink-200 h-8"><span class="whitespace-nowrap">
        text-prefix
      </span></div>
  <zinput-stub name="" value="" label="" type="text" placeholder="Enter a value" debouncedelay="350" size="small" backgroundcolor="ink-400" showborder="true" errorbordercolor="cherry" class="rounded-tl-none rounded-bl-none"></zinput-stub>
</div>
`;

exports[`InputGroup Component renders an input element with addon 1`] = `
<div class="flex">
  <div class="flex items-center p-2 rounded-tl-sm rounded-bl-sm outline-none bg-ink-200 h-10"><span class="whitespace-nowrap">
        text-prefix
      </span></div>
  <zinput-stub name="" value="" label="" type="text" placeholder="Enter a value" debouncedelay="350" size="medium" backgroundcolor="ink-400" showborder="true" errorbordercolor="cherry" class="rounded-tl-none rounded-bl-none"></zinput-stub>
</div>
`;

exports[`InputGroup Component renders an input element with addon having a different background color 1`] = `
<div class="flex">
  <div class="flex items-center p-2 rounded-tl-sm rounded-bl-sm outline-none bg-juniper h-10"><span class="whitespace-nowrap">
        text-prefix
      </span></div>
  <zinput-stub name="" value="" label="" type="text" placeholder="Enter a value" debouncedelay="350" size="medium" backgroundcolor="ink-400" showborder="true" errorbordercolor="cherry" class="rounded-tl-none rounded-bl-none"></zinput-stub>
</div>
`;
